Soccar [UI]/src/nl/soccar/ui/input/InputController.java

```diff
@@ -0,0 +1,109 @@
+package nl.soccar.ui.input;
+
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import net.java.games.input.Controller;
+import net.java.games.input.ControllerEnvironment;
+import nl.soccar.library.enumeration.HandbrakeAction;
+import nl.soccar.library.enumeration.SteerAction;
+import nl.soccar.library.enumeration.ThrottleAction;
+
+/**
+ *
+ * @author PTS34A
+ */
+public final class InputController {
+
+    private static final InputController INSTANCE = new InputController();
+
+    private ThrottleAction throttleAction;
+    private HandbrakeAction handbrakeAction;
+    private SteerAction steerAction;
+
+    private Keyboard keyboard;
+    private GamePad gamePad;
+    private final Timer timer = new Timer(true);
+
+    public void initializeInput(Keyboard keyboard) {
+        this.keyboard = keyboard;
+        start();
+    }
+
+    private void start() {
+        timer.scheduleAtFixedRate(new TimerTask() {
+            @Override
+            public void run() {
+                poll();
+            }
+        }, 0, 50);
+    }
+
+    private void poll() {
+        if (gamePad == null) {
+            pollKeyboard();
+        } else if (gamePad.isConnected()) {
+            setThrottleAction(gamePad.getThrottleAction());
+            setHandbrakeAction(gamePad.getHandbrakeAction());
+            setSteerAction(gamePad.getSteerAction());
+        } else {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 45. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1871)